stestr/repository/util.py

```diff
@@ -22,26 +24,42 @@ def _get_default_repo_url(repo_type):
     return repo_url
 
 
-def get_repo_open(repo_type, repo_url=None):
+def get_repo_open(repo_type=None, repo_url=None):
     """Return an already initialized repo object given the parameters
 
-    :param str repo_type: The repo module to use for the returned repo
+    :param str repo_type: DEPRECATED - The repo module to use for the returned
+        repo
     :param str repo_url: An optional repo url, if one is not specified the
         default $CWD/.stestr will be used.
     """
+    if repo_type is not None:
+        msg = ("WARNING: Specifying repository type is deprecated and will be "
+               "removed in future release.\n")
+        sys.stderr.write(msg)
```

Comment:
I'm not sure we need to write the warning to stderr here. We were doing that for the CLI to make sure users saw the warning as the `DeprecationWarning` is normally not seen. But since the repo type argument no longer exists on the CLI this is only for python library interface users and I think relying on the warnings interface may be enough for them. What do you think?

Reply:
well, yeah, indeed. I'll remove this one and `get_repo_initilise()`'s one, too.